pom.xml

```diff
@@ -9,8 +9,10 @@
     <packaging>jar</packaging>
 
     <properties>
-        <maven.compiler.source>22</maven.compiler.source>
-        <maven.compiler.target>22</maven.compiler.target>
```

Comment:
why are we switching to java 23 here? do we know this doesn't introduce any weird behaviour with the other dependencies/everyone else's setup?